src/scales/scale.linearbase.js

```diff
@@ -16,35 +16,41 @@ function generateTicks(generationOptions, dataRange) {
 	// for details.
 
 	var stepSize = generationOptions.stepSize;
```

Comment:
`stepSize < 0` doesn't seems a valid value, so maybe we can simplify all these checks by clamping this value:

```javascript
var stepSize = Math.max(0, generationOptions.stepSize);
var unit = stepSize || 1;
//...
```